data/summary/types/summary.go

```diff
@@ -161,7 +168,7 @@ type StatsPt[T Stats] interface {
 	Init()
 	GetBucketsLen() int
 	GetBucketDate(int) time.Time
-	Update(context.Context, DeviceDataCursor) error
+	Update(context.Context, DeviceDataCursor, DeviceDataManager) error
```

Comment:
Is there a way this can be simplified to only use the device data manager instead of a cursor and a manager?